### PR TITLE
Tapis headers

### PR DESCRIPTION
**Pull request by richcar58**
This pull request contains the SK code that implements explicit tenant/user parameter passing.

### CODE REVIEW

File: tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java [hunk 1]
```diff
@@ -1,7 +1,34 @@
...
+    public String tenant;
+    public String user;
     public String parentRoleName;
     public String childRoleName;
+    
+    /** Return a user-appropriate error message on failed validation
+     *  and return null if validation succeeds.
+     */ 
+    @Override
+    public String validate()
```
Comment (reptillicus) [inline tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java]: Any reason not to use the javax.validations @NotNull annotations?

File: tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java [hunk 2]
```diff
@@ -1,7 +1,34 @@
...
+        // Final checks.
+        if (StringUtils.isBlank(tenant)) 
+            return MsgUtils.getMsg("TAPIS_NULL_PARAMETER", "addChildRole", "tenant");
+        if (StringUtils.isBlank(user)) 
+            return MsgUtils.getMsg("TAPIS_NULL_PARAMETER", "addChildRole", "user");
+        if (StringUtils.isBlank(parentRoleName)) 
+            return MsgUtils.getMsg("TAPIS_NULL_PARAMETER", "addChildRole", "parentRoleName");
+        if (StringUtils.isBlank(childRoleName)) 
+            return MsgUtils.getMsg("TAPIS_NULL_PARAMETER", "addChildRole", "childRoleName");
+
```
Comment (reptillicus) [inline tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java]: If there are multiple validation errors would it be possible to send them all back at once?

File: tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java (hunk 1 above)
Comment (richcar58) [reply tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java]: Annotations an alternative way of doing things.  One of the downsides is that checking for empty strings is almost always also required.  I'm also not sure what type of error messages get generated using the annotation.  It's worth considering.

File: tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java (hunk 2 above)
Comment (richcar58) [reply tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java]: It's only a matter of coding.

File: tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java (hunk 1 above)
Comment (reptillicus) [reply tapis-securityapi/src/main/java/edu/utexas/tacc/tapis/security/api/requestBody/ReqAddChildRole.java]: There is also @NotEmpty and @NotBlank

**Pull request merged**